core/src/bootstrap.rs

```diff
@@ -0,0 +1,766 @@
+            BootstrapError::YamlRust("`wallet_address_for_chain_transaction` missing".into())
+        })?;
+    let wallet_address_for_chain_transaction = AddressEncoder::new(network_prefix)
+        .parse_address_from_str(wallet_address_for_chain_transaction_str)?;
+
+    let addresses = Addresses {
+        address_for_oracle_tokens: address_for_minted_tokens,
+        wallet_address_for_chain_transaction,
+    };
+    let refresh_contract_parameters_str = yaml["refresh_contract_parameters"]
+        .as_str()
+        .ok_or_else(|| BootstrapError::YamlRust("`refresh_contract_parameters` missing".into()))?;
+    let refresh_contract_parameters: RefreshContractParameters =
+        serde_yaml::from_str(refresh_contract_parameters_str)?;
+    let node_ip = yaml["node_ip"]
+        .as_str()
+        .ok_or_else(|| BootstrapError::YamlRust("`node_ip` missing".into()))?
+        .into();
+
+    let node_port = yaml["node_port"]
+        .as_str()
+        .ok_or_else(|| BootstrapError::YamlRust("`node_port` missing".into()))?
+        .into();
+
+    let node_api_key = yaml["node_api_key"]
+        .as_str()
+        .ok_or_else(|| BootstrapError::YamlRust("`node_api_key` missing".into()))?
+        .into();
+
+    let tokens_to_mint: TokensToMint = serde_yaml::from_str(tokens_to_mint_str)?;
+
+    Ok(BootstrapConfig {
+        refresh_contract_parameters,
+        tokens_to_mint,
+        node_ip,
+        node_port,
+        node_api_key,
+        is_mainnet,
+        addresses,
+    })
+}
+
+/// An instance of this struct is created from an operator-provided YAML file.
+pub struct BootstrapConfig {
+    pub refresh_contract_parameters: RefreshContractParameters,
+    pub tokens_to_mint: TokensToMint,
+    pub node_ip: String,
+    pub node_port: String,
+    pub node_api_key: String,
+    pub is_mainnet: bool,
+    pub addresses: Addresses,
+}
+
+pub struct Addresses {
+    pub address_for_oracle_tokens: Address,
+    pub wallet_address_for_chain_transaction: Address,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct TokensToMint {
+    pub pool_nft: NftMintDetails,
+    pub refresh_nft: NftMintDetails,
+    pub update_nft: NftMintDetails,
+    pub oracle_tokens: TokenMintDetails,
+    pub ballot_tokens: TokenMintDetails,
+    pub reward_tokens: TokenMintDetails,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct RefreshContractParameters {
+    pub epoch_length: u32,
+    pub buffer: u32,
+    pub total_oracles: u32,
+    pub min_data_points: u32,
+    pub max_deviation_percent: u32,
+    pub total_ballots: u32,
+    pub min_votes: u32,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct TokenMintDetails {
+    pub name: String,
+    pub description: String,
+    pub quantity: u64,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct NftMintDetails {
+    pub name: String,
+    pub description: String,
+}
+
+#[derive(Serialize)]
+pub struct OracleConfigFields {
+    #[serde(serialize_with = "token_id_as_base64_string")]
```

Comment:
Whoa! I totally missed that there is serde-yaml. I wonder how could it simplify yaml parsing in my typed oracle config PR.

Reply:
I think there could definitely be chunks where you can defer to `serde-yaml`. I couldn't use it for the top level bootstrap config because I needed to first parse `is_mainnet`, to check that the addresses are valid. If you separate out chunks into sub-structs where you don't need to do that kind of processing, then we can offload it.